src/Workspaces/Core/Portable/Options/GlobalOptionService.cs

```diff
@@ -91,12 +95,28 @@ static ImmutableHashSet<IOption> ComputeSerializableOptionsFromProviders(Immutab
             }
         }
 
+        private ImmutableArray<IOptionPersister> GetOptionPersisters()
+        {
+            if (_lazyOptionSerializers is { IsDefault: false } optionSerializers)
+            {
+                return optionSerializers;
+            }
+
+            var constructedOptionSerializers = _optionSerializerProviders.SelectAsArray(
+                static (lazyProvider) => lazyProvider.Value.TryGetPersister());
+
+            optionSerializers = constructedOptionSerializers.WhereNotNull().ToImmutableArray();
+            ImmutableInterlocked.InterlockedInitialize(ref _lazyOptionSerializers, optionSerializers);
+
+            return _lazyOptionSerializers;
```

Comment:
can you doc this method?  i don't quite get the semantics of using TryGetPersistor here.  it seems like it means we can cache and hold onto an intermediary state where only some persisters had been initialized and some had not.  i don't really get it.

Reply:
Sure will doc

Reply:
(or just try opening a CPS based project...)

Reply:
> So I suspect the behavior where the global option service is caching a list, and we drop uncreated ones permanently is going to be broken.

This is now rewritten to remove the requirement that the persisters be initialized before calling this method.